library/alloc/src/fmt.rs

```diff
@@ -604,9 +604,15 @@ use crate::string;
 #[cfg(not(no_global_oom_handling))]
 #[must_use]
 #[stable(feature = "rust1", since = "1.0.0")]
+#[inline]
 pub fn format(args: Arguments<'_>) -> string::String {
-    let capacity = args.estimated_capacity();
-    let mut output = string::String::with_capacity(capacity);
-    output.write_fmt(args).expect("a formatting trait implementation returned an error");
-    output
+    #[cold]
```

Comment:
`format_cold` is the most common path, I don't think we want this to be `#[cold]`.

Reply:
It doesn't effect the output as far as I can tell. Current `format` gets implemented cold too (I've not been able to get any code that does not make a call to format in the asm)